internal/hcs/schema2/virtual_machine_memory.go

```diff
@@ -0,0 +1,59 @@
+	// If enabled, then each backing page is physically pinned on first access.
+	PinBackingPages bool `json:"PinBackingPages,omitempty"`
+	// If enabled, then backing page chunks smaller than the backing page size are never used unless the system is under extreme memory pressure. If the backing page size is Small, then it is forced to Large when this option is enabled.
+	ForbidSmallBackingPages       bool `json:"ForbidSmallBackingPages,omitempty"`
+	EnablePrivateCompressionStore bool `json:"EnablePrivateCompressionStore,omitempty"`
+	// If enabled, then the memory hot hint feature is exposed to the VM, allowing it to prefetch pages into its working set. (if supported by the guest operating system).
+	EnableHotHint bool `json:"EnableHotHint,omitempty"`
+	// If enabled, then the memory cold hint feature is exposed to the VM, allowing it to trim zeroed pages from its working set (if supported by the guest operating system).
+	EnableColdHint bool `json:"EnableColdHint,omitempty"`
+	// If enabled, then the memory cold discard hint feature is exposed to the VM, allowing it to trim non-zeroed pages from the working set (if supported by the guest operating system).
+	EnableColdDiscardHint bool `json:"EnableColdDiscardHint,omitempty"`
+	// If enabled, then the base address of direct-mapped host images is exposed to the guest.
+	ImageBaseAddressesExposed  bool     `json:"ImageBaseAddressesExposed,omitempty"`
+	SharedMemoryMB             int64    `json:"SharedMemoryMB,omitempty"`
+	DisableSharedMemoryMapping bool     `json:"DisableSharedMemoryMapping,omitempty"`
+	SharedMemoryAccessSids     []string `json:"SharedMemoryAccessSids,omitempty"`
+	EnableEpf                  bool     `json:"EnableEpf,omitempty"`
+	// If enabled, then commit is not charged for each backing page until first access.
+	EnableDeferredCommit bool `json:"EnableDeferredCommit,omitempty"`
+	// Low MMIO region allocated below 4GB
+	LowMMIOGapInMB uint64 `json:"LowMmioGapInMB,omitempty"`
+	// High MMIO region allocated above 4GB (base and size)
+	HighMMIOBaseInMB         uint64                 `json:"HighMmioBaseInMB,omitempty"`
+	HighMMIOGapInMB          uint64                 `json:"HighMmioGapInMB,omitempty"`
+	Vtl2AddressRangeBase     uint64                 `json:"Vtl2AddressRangeBase,omitempty"`
+	Vtl2AddressRangeSize     uint64                 `json:"Vtl2AddressRangeSize,omitempty"`
+	Vtl2MMIOAddressRangeSize uint64                 `json:"Vtl2MmioAddressRangeSize,omitempty"`
+	Priority                 uint32                 `json:"Priority,omitempty"`
+	LimitInMB                uint64                 `json:"LimitInMB,omitempty"`
+	ReservationInMB          uint64                 `json:"ReservationInMB,omitempty"`
+	TargetBuffer             uint32                 `json:"TargetBuffer,omitempty"`
+	// HCS currently doesn't support DM enable, will only have value for VMMS VM
+	EnableDynamicMemory        bool                               `json:"EnableDynamicMemory,omitempty"`
```

Comment:
We should remove these unused fields as well

Reply:
Done.